amqp/src/main/scala/akka/stream/alpakka/amqp/impl/AmqpAsyncFlow.scala

```diff
@@ -0,0 +1,332 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.amqp.impl
+
+import akka.Done
+import akka.annotation.InternalApi
+import akka.event.Logging
+import akka.stream._
+import akka.stream.alpakka.amqp.impl.AmqpAsyncFlowStageLogic.DeliveryTag
+import akka.stream.alpakka.amqp.{AmqpWriteSettings, WriteMessage, WriteResult}
+import akka.stream.stage._
+import com.rabbitmq.client.ConfirmCallback
+
+import scala.collection.immutable.TreeMap
+import scala.collection.mutable
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{Future, Promise}
+
+/**
+ * Internal API.
+ */
+@InternalApi private final case class AwaitingMessage[T](
+    tag: DeliveryTag,
+    passThrough: T,
+    var ready: Boolean = false
```

Comment:
does it have to be var? I see that in `setReady` the object is copied, not modified.